Sources/HttpPipeline/SignedCookies.swift

```diff
@@ -0,0 +1,148 @@
+    let decrypt = !signedCookieValue.contains("--")
+
+    guard let cookieValue = decrypt ? decrypted(text: signedCookieValue, secret: secret) : signedCookieValue
+      else { return nil }
+
+    let parts = cookieValue.components(separatedBy: "--")
+    guard let encodedValue = parts.first,
+      let providedDigest = parts.last
+      else {
+        return nil
+    }
+
+    let trueDigest = digest(value: encodedValue, secret: secret)
+
+    guard trueDigest == providedDigest else { return nil }
+    guard let value = base64DecodedData(string: encodedValue) else { return nil }
+
+    return .some(value)
+  }
+
+  /// Helper function that calls `verifiedData` and then tries converting the data to a string.
+  public static func verifiedString(signedCookieValue: String, secret: String)
+    -> String? {
+      return verifiedData(signedCookieValue: signedCookieValue, secret: secret)
+        .flatMap { String(data: $0, encoding: .utf8) }
+  }
+
+  /// Help function that calls `verifiedData` and then tries to decode the data into an `A`.
+  public static func verifiedValue<A: Decodable>(signedCookieValue: String, secret: String) -> A? {
+      return verifiedData(signedCookieValue: signedCookieValue, secret: secret)
+        .flatMap { try? JSONDecoder().decode(A.self, from: $0) }
+  }
+}
+
+private func digest(value: String, secret: String) -> String? {
+  let keyBytes = CryptoUtils.byteArray(fromHex: secret)
+  let valueBytes = CryptoUtils.byteArray(from: value)
+  let digestBytes = HMAC(using: .sha256, key: keyBytes).update(byteArray: valueBytes)?.final()
+  return digestBytes.map { Data(bytes: $0).base64EncodedString() }
+}
+
+private func base64DecodedData(string: String) -> Data? {
+  return Data(base64Encoded: Data(string.utf8))
+}
+
+private func encrypted(text plainText: String, secret: String) -> String? {
+  let secretBytes = CryptoUtils.byteArray(fromHex: secret)
+  let iv = [UInt8](repeating: 0, count: secretBytes.count)
+  let plainTextBytes = CryptoUtils.byteArray(from: plainText)
+
+  let blockSize = Cryptor.Algorithm.aes.blockSize
+  let paddedPlainTextBytes = plainTextBytes.count % blockSize != 0
+    ? CryptoUtils.zeroPad(byteArray: plainTextBytes, blockSize: blockSize)
+    : plainTextBytes
+
+  let cipherText = Cryptor(operation: .encrypt, algorithm: .aes, options: .none, key: secretBytes, iv: iv)
+    .update(byteArray: paddedPlainTextBytes)?
+    .final()
+
+  return cipherText.map { CryptoUtils.hexString(from: $0) }
+}
+
+private func decrypted(text encryptedText: String, secret: String) -> String? {
+  let secretBytes = CryptoUtils.byteArray(fromHex: secret)
+  let iv = [UInt8](repeating: 0, count: secret.count)
+  let encryptedTextBytes = CryptoUtils.byteArray(fromHex: encryptedText)
+
+  let decryptedText = Cryptor(operation: .decrypt, algorithm: .aes, options: .none, key: secretBytes, iv: iv)
+    .update(byteArray: encryptedTextBytes)?
+    .final()
+
+  return decryptedText
+    .map { Data($0.filter { $0 != 0 }) }
```

Comment:
gotta filter out `0` from the decrypted bytes since we padded in the `encrypted` function...